### ISSUE TITLE
[BUG] Inconsistency with trailing slashes with API URLs in documentation sample code, 404 when trailing slash is present

### ISSUE THREAD

**Issue by ILW8** [uclapi/uclapi#2964]
**Describe the bug**
The URLs provided in the sample code in the API Documentation [sometimes contain trailing slashes](https://uclapi.com/docs/#dashboard/api/analytics/services). When running the sample code, we would get 404 because of the trailing slash, whereas it would work if the trailing slash is removed from the URL.

**To Reproduce**
Steps to reproduce the behavior (using one of the endpoints):
1. request https://uclapi.com/dashboard/api/analytics/methods/
2. Get 404

**Expected behavior**
A response from the API (not a 404)

**Additional context**
Django usually uses trailing slashes with all their URLs, however there is significant controversy regarding whether an URI with a trailing slash is the same as another URI _without_ a trailing slash ([discussion 1](https://stackoverflow.com/questions/61547014/restful-uri-trailing-slash-or-no-trailing-slash), [discussion 2](https://github.com/open-policy-agent/opa/issues/972#issuecomment-427937639)). 

By default, Django [has a setting that redirects URLs](https://docs.djangoproject.com/en/dev/ref/settings/?from=olddocs#append-slash) that don't match any urlconf to the same URL plus a trailing slash. Adding a trailing slash to the URL patterns in the respective `urls.py` would likely solve the issue, however the documentation states that:
> Note that the redirect **may** cause any data submitted in a POST request to be lost.

and I'm not sure how that affects the endpoints that do use POST. @greenfrogs

**Comment by ChristopherHammond13** [comment]
I hope I may weigh in here! I would make the argument that we should not require the final slash `/`. I have a few reasons for this, some technical, some not.
1) Ending in a slash, to me, says "hey, I got this far but I'm not really sure what I wanted to type next". When you make an API request, you are making a definitive query or requesting a definitive action; your foot isn't half in so to speak.
2) Most (all, still?) of this API is all about obtaining information. Therefore, even if you're, for example, grabbing all `/rooms`, you are expecting specifying one endpoint, which is to return a collection. There is nothing below `/rooms`; it is the deepest level, so I don't think it makes sense to end it in a `/`.
3) Usually when you navigate to a URL ending in a `/`, and you get a result anyway, it's because something ✨ magical ✨ has happened behind the scenes. Usually, this act of magic 🧙 is a web server *rewriting* that URL to secretly return you some index page, such as `index.html`, `index.php` or `default.aspx`. This is an API, not a website, and such we should not set the expectation that we will do any kind of rewrite magic. An API is all about a transaction; we shouldn't play guess work (in my opinion) to dry to figure out what the user meant.

This answers whether I think we should end things in a slash. Now, the next question is "should we allow it?". And on that I say: it depends. When building a service, and especially one for whom the target audience is students we want to inspire and empower, we shouldn't try to be annoyingly finnicky. We don't want to put off the next generation because they dared to send us a `/`. We also don't want to box ourselves into a corner either, if we decided to have (for example) `/rooms/{room_id}`.

There are roughly three options:
1) Fix docs and deny the trailing slash because it's wrong (reasonable. Harsh for somebody who makes that error, but reasonable if they did something the docs told them not to). Make sure this is documented.
2) Allow it, and hope for the best later. Ensure that, whenever new endpoints are made, undefined behaviour or quirks won't occur in either case.
3) Allow it, but provide a friendly warning in the JSON.

That's not my decision to make any more, but thought I'd throw my two pence onto the pile. Good luck!
